airflow/providers/microsoft/azure/hooks/asb.py

```diff
@@ -0,0 +1,202 @@
+    """
+    BaseAzureServiceBusHook class to create session and create connection using connection string
+
+    :param azure_service_bus_conn_id: Reference to the
+        :ref:`Azure Service Bus connection<howto/connection:azure_service_bus>`.
+    """
+
+    conn_name_attr = 'azure_service_bus_conn_id'
+    default_conn_name = 'azure_service_bus_default'
+    conn_type = 'azure_service_bus'
+    hook_name = 'Azure Service Bus'
+
+    @staticmethod
+    def get_ui_field_behaviour() -> Dict[str, Any]:
+        """Returns custom field behaviour"""
+        return {
+            "hidden_fields": ['port', 'host', 'extra', 'login', 'password'],
+            "relabeling": {'schema': 'Connection String'},
+            "placeholders": {
+                'schema': 'Endpoint=sb://<Resource group>.servicebus.windows.net/;SharedAccessKeyName=<AccessKeyName>;SharedAccessKey=<SharedAccessKey>',  # noqa
+            },
+        }
+
+    def __init__(self, azure_service_bus_conn_id: str = default_conn_name) -> None:
+        super().__init__()
+        self.conn_id = azure_service_bus_conn_id
+
+    def get_conn(self):
+        raise NotImplementedError
+
+
+class AdminClientHook(BaseAzureServiceBusHook):
+    """
+    Interacts with ServiceBusAdministrationClient client
+    to create, update, list, and delete resources of a
+    Service Bus namespace.  This hook uses the same Azure Service Bus client connection inherited
+    from the base class
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    def get_conn(self) -> ServiceBusAdministrationClient:
+        """
+        Create and returns ServiceBusAdministrationClient by using the connection
+        string in connection details
+        """
+        conn = self.get_connection(self.conn_id)
+
+        connection_string: str = str(conn.schema)
+        return ServiceBusAdministrationClient.from_connection_string(connection_string)
+
+    def create_queue(
+        self,
+        queue_name: str,
+        max_delivery_count: int = 10,
+        dead_lettering_on_message_expiration: bool = True,
+        enable_batched_operations: bool = True,
+    ) -> QueueProperties:
+        """
+        Create Queue by connecting to service Bus Admin client return the QueueProperties
+
+        :param queue_name: The name of the queue or a QueueProperties with name.
+        :param max_delivery_count: The maximum delivery count. A message is automatically
+            dead lettered after this number of deliveries. Default value is 10..
+        :param dead_lettering_on_message_expiration: A value that indicates whether this subscription has
+            dead letter support when a message expires.
+        :param enable_batched_operations: Value that indicates whether server-side batched
+            operations are enabled.
+        """
+        if queue_name is None:
+            raise ValueError("Queue name cannot be None.")
```

Comment:
```suggestion
            raise TypeError("Queue name cannot be None.")
```
Looks like there is a mix of `TypeError` and `ValueError` for the same exception. Let's stay consistent with Ash's initial feedback.

Reply:
Moving back to raise `TypeError`